core/revapi.json

```diff
@@ -4830,6 +4830,20 @@
         "new": "method java.util.UUID com.datastax.oss.driver.api.core.metadata.Node::getHostId()",
         "annotation": "@edu.umd.cs.findbugs.annotations.Nullable",
         "justification": "JAVA-2505: Annotate Node.getHostId() as nullable"
+      },
+      {
+        "code": "java.annotation.added",
+        "old": "parameter void com.datastax.oss.driver.api.core.ssl.ProgrammaticSslEngineFactory::<init>(===javax.net.ssl.SSLContext===)",
+        "new": "parameter void com.datastax.oss.driver.api.core.ssl.ProgrammaticSslEngineFactory::<init>(===javax.net.ssl.SSLContext===)",
+        "annotation": "@edu.umd.cs.findbugs.annotations.NonNull",
+        "justification": "JAVA-2434: added @NonNull to ProgrammaticSslEngineFactory(SSLContext) constructor"
```

Comment:
can we introduce not backward-compatible changes?

Reply:
In this case I need to remove the `@NonNull` annotation. But honestly adding/removing annotations doesn't really break users code.

Reply:
Hmm you are right for the `throws Exception` clause, this could break user code. I will remove it.